Sofa/Component/Mapping/Linear/src/sofa/component/mapping/linear/BarycentricMappers/BarycentricMapperMeshTopology.inl

```diff
@@ -713,85 +713,85 @@ const sofa::linearalgebra::BaseMatrix* BarycentricMapperMeshTopology<In,Out>::ge
     const SeqTetrahedra& tetrahedra = this->m_fromTopology->getTetrahedra();
     const SeqHexahedra& cubes = this->m_fromTopology->getHexahedra();
 
-
+    int i {};
     // 1D elements
     {
-        for ( size_t i=0; i<m_map1d.size(); i++ )
+        for ( const auto& map1d : m_map1d)
         {
-            const size_t out = i;
-            const Real fx = ( Real ) m_map1d[i].baryCoords[0];
-            size_t index = m_map1d[i].in_index;
+            const auto fx = map1d.baryCoords[0];
+            const auto index = map1d.in_index;
             {
                 const Edge& line = lines[index];
-                this->addMatrixContrib(m_matrixJ, out, line[0],  ( 1-fx ));
-                this->addMatrixContrib(m_matrixJ, out, line[1],  fx);
+                this->addMatrixContrib(m_matrixJ, i, line[0],  1 - fx);
+                this->addMatrixContrib(m_matrixJ, i, line[1],  fx);
             }
+            ++i;
         }
     }
     // 2D elements
     {
-        const size_t i0 = m_map1d.size();
         const size_t c0 = triangles.size();
-        for ( size_t i=0; i<m_map2d.size(); i++ )
+        for ( const auto& map2d : m_map2d)
         {
-            const size_t out = i+i0;
-            const Real fx = ( Real ) m_map2d[i].baryCoords[0];
-            const Real fy = ( Real ) m_map2d[i].baryCoords[1];
-            size_t index = m_map2d[i].in_index;
-            if ( index<c0 )
+            const Real fx = map2d.baryCoords[0];
+            const Real fy = map2d.baryCoords[1];
+            const size_t index = map2d.in_index;
+            if ( index < c0 )
             {
                 const Triangle& triangle = triangles[index];
-                this->addMatrixContrib(m_matrixJ, out, triangle[0],  ( 1-fx-fy ));
-                this->addMatrixContrib(m_matrixJ, out, triangle[1],  fx);
-                this->addMatrixContrib(m_matrixJ, out, triangle[2],  fy);
+                this->addMatrixContrib(m_matrixJ, i, triangle[0],  ( 1-fx-fy ));
```

Comment:
It took me some time to understand the logic behing  the 'i' but I got it and seems ok.